tests/test_docrep.py

```diff
@@ -592,6 +592,24 @@ def test_delete_kwargs(self):
         with self.assertWarns(UserWarning):
             self.ds.delete_kwargs('test')
 
+    @unittest.skipIf(not six.PY2, "Only implemented for python 2.7")
+    def test_py2_classes(self):
+        """Test the handling of classes in python 2.7"""
+        # this should work
+        @self.ds
+        class Test(object):
+            """docs"""
+        # this should not
+        self.ds.python2_classes = 'raise'
+        try:
+            @self.ds
+            class Test2(object):
+                """docs"""
+        except AttributeError:
```

Comment:
Just a side-comment (not saying that you should do it in this PR at all) but with pytest you can check expected exception like this:

```py
with pytest.raises(AttributeError, 'the regex to check):
    # code that raises
```

See [this](https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions) for more details.

Reply:
Yep I know. I usually force myself to use the unittest framework only and there is no such thing for python 2, only for python 3.

Reply:
> I usually force myself to use the unittest framework only

Interesting, do you mind to elaborate why?

Reply:
Sure. I mainly base my tests on unittest because it integrates well with the object-oriented structure of my packages. Additionally, since it is a built-in library, it is very stable and keeps my dependencies for running the tests as low. Anyway, of course I use pytest as well, since it provides a better interfaces for running the tests. But if it is only for a minor thing (like here, where I would only use it for this single test method) I'd rather not rely on non built-in libraries. But for other packages with more extensive test suites, I indeed often rely on pytest but still use unittest as a basis

Reply:
OK thanks for the details. I have to say that there are so many nice things included in pytest both for writing tests and running the tests that you would have to pry it from my cold dead hands ...

Reply:
😆 agreed 😜